packages/target-tauri/src-tauri/src/util/fs_watcher.rs

```diff
@@ -0,0 +1,66 @@
+use std::{
+    future::Future,
+    path::Path,
+    time::{Duration, SystemTime},
+};
+
+use notify::{Config, Event, RecommendedWatcher, RecursiveMode, Watcher};
+use tauri::async_runtime::{block_on, channel, Receiver};
+
+fn async_watcher() -> notify::Result<(RecommendedWatcher, Receiver<notify::Result<Event>>)> {
+    let (tx, rx) = channel(1);
+
+    let watcher = RecommendedWatcher::new(
+        move |res| {
+            block_on(async {
+                tx.send(res).await.unwrap();
+            })
+        },
+        Config::default(),
+    )?;
+
+    Ok((watcher, rx))
+}
+
+pub async fn async_watch_debounced<P, F>(
+    path: P,
+    callback: Box<dyn Fn() -> F + Sync + Send>,
+    delay: Duration,
+) -> notify::Result<()>
+where
+    P: AsRef<Path>,
+    F: Future<Output = ()>,
+{
+    let (mut watcher, mut rx) = async_watcher()?;
+
+    // Add a path to be watched. All files and directories at that path and
+    // below will be monitored for changes.
+    watcher.watch(path.as_ref(), RecursiveMode::Recursive)?;
+
+    let mut last_action = SystemTime::now();
+
+    while let Some(res) = rx.recv().await {
+        match res {
+            Ok(event) => {
+                log::trace!("changed: {:?}", event);
+                if event.kind.is_create() || event.kind.is_modify() || event.kind.is_remove() {
+                    let now = SystemTime::now();
```

Comment:
maybe we should replace it with `Instant::now()` because `SystemTime` can run backwards in theory.

Reply:
Though on the other hand instant is a bit counter intuitive as time for it is not linear.. We gotta try it to be sure.